### PR TITLE
workflows/actionlint: run `zizmor`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Port of Homebrew/homebrew-core#195961.

See https://github.com/woodruffw/zizmor.

### CODE REVIEW

File: .github/workflows/actionlint.yml [hunk 1]
```diff
@@ -29,27 +29,45 @@ concurrency:
...
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/homebrew/ubuntu22.04:master
     steps:
       - name: Set up Homebrew
         id: setup-homebrew
         uses: Homebrew/actions/setup-homebrew@master
         with:
-          core: false
+          core: true
```
Comment (ZhongRuoyu) [inline .github/workflows/actionlint.yml:40]: Is the core tap needed?

If this change is not intentional then you may also want to relook at the change from `HOMEBREW_REPOSITORY` to `HOMEBREW_TAP_REPOSITORY` below.

Comment (carlocab) [reply .github/workflows/actionlint.yml:39]: Oops, thanks

Comment (carlocab) [reply .github/workflows/actionlint.yml]: Fixed!

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @carlocab!

**Pull request merged**